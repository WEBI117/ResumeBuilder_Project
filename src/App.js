import React, { useState, useEffect } from 'react';
import Logincomponent from './components/Logincomponent/login'
import Resumebuilder from './components/Resumebuildercomponent/resumebuilder'
import 'bootstrap/dist/css/bootstrap.min.css';
function App(props) {
    let loginStatusUpdateEvent = (e,status) => {
        updateLoginStatus(status)
    }
    // react hook to track state for application login status.
    let [isloggedin,updateLoginStatus]=useState(props.loginstatus);
    // component placeholder determined by login status. Will display login page incase user has not logged in already.
    let [content,setContent] = useState(<Logincomponent loginStatusHandler={loginStatusUpdateEvent}/>);

    useEffect(() => {
        if(!isloggedin){
            setContent(<Logincomponent loginStatusHandler={loginStatusUpdateEvent}/>)
        }
        else{
            setContent(<Resumebuilder loginStatusUpdateEvent={loginStatusUpdateEvent}/>)
        }
    },[isloggedin])

    // function to be passed to children page components to trigger loggin status changes in parent component.
    // 

    //if(!isloggedin){
    //    content=<Logincomponent loginStatusHandler={loginStatusUpdateEvent}/>
    //}
    //else{
    //    content=<Resumebuilder/>
    //}
  return (
    <div className="App">
      {content}
    </div>
  );
}

export default App;
