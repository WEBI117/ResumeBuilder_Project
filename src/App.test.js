import ReactDOM from 'react-dom/client'
import { render, screen , cleanup} from '@testing-library/react';
import App from './App';
import TestRenderer from 'react-test-renderer';
import {create,act} from 'react-test-renderer';
import "@testing-library/jest-dom/extend-expect"

afterEach(cleanup)
test('Application is rendered', () => {
    const div = document.createElement('div');
    const root = ReactDOM.createRoot(div)
})

test('Application login tab renders correctly', ()=> {
    const tree = TestRenderer.create(<App loginstatus={false}/>).toJSON()
    expect(tree).toMatchSnapshot()
})

