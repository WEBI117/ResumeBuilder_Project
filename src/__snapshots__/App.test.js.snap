// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application login tab renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="login"
    title="logincomponent"
  >
    <div
      className="container"
    >
      <div
        className="col-md-2 offset-md-5"
      >
        <div
          className="row"
          title="emailinput"
        >
          <div
            className="col-md-2"
          >
            <div
              className="row"
            >
              <div>
                <label
                  className="label"
                  htmlFor="email"
                >
                  Email: 
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div>
                <input
                  className="input"
                  data-testid="email"
                  id="email-input"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
          title="passwordinput"
        >
          <div
            className="col-md-2"
          >
            <div
              className="row"
            >
              <div>
                <label
                  className="label"
                  htmlFor="pass"
                >
                  Password: 
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div>
                <input
                  className="input"
                  data-testid="pass"
                  id="password-input"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "height": "20px",
            }
          }
        />
        <div
          className="row"
        >
          <div
            className="col-md-1"
          >
            <button
              disabled={true}
              id="loginbutton"
              onClick={[Function]}
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
